contrib/fuzz/container_fuzzer.go

```diff
@@ -38,11 +38,69 @@ import (
 	"time"
 )
 
-func init() {
-	err := updatePathEnv()
-	if err != nil {
-		fmt.Println(err)
+var (
+	haveInstalledWget      = false
+	haveDownloadedbinaries = false
+	haveExtractedBinaries  = false
+	haveChangedPATH        = false
+	haveInitialized        = false
+)
+
+// initInSteps() performs initialization in several steps
+// The reason for spreading the initialization out in
+// multiple steps is that each fuzz iteration can maximum
+// take 25 seconds when running through OSS-fuzz.
+// Should an iteration exceed that, then the fuzzer stops.
+func initInSteps() bool {
+	// Install wget
+	if !haveInstalledWget {
+		cmd := exec.Command("apt-get", "install", "-y", "wget")
+		err := cmd.Run()
+		if err != nil {
+			return true
+		}
+		haveInstalledWget = true
+		return true
+	}
+	// Download binaries
+	if !haveDownloadedbinaries {
+		tarLink := "https://github.com/containerd/containerd/releases/download/v1.5.4/containerd-1.5.4-linux-amd64.tar.gz"
+		cmd := exec.Command("wget", tarLink)
+		err := cmd.Run()
+		if err != nil {
+			return true
+		}
+		haveDownloadedbinaries = true
+		return true
+	}
+	// Extract binaries
+	binariesDir := "/tmp/containerd-binaries"
+	if !haveExtractedBinaries {
+		err := os.MkdirAll(binariesDir, 0777)
+		if err != nil {
+			return true
+		}
+		cmd := exec.Command("tar", "xvf", "containerd-1.5.4-linux-amd64.tar.gz", "-C", binariesDir)
+		err = cmd.Run()
+		if err != nil {
+			return true
+		}
+		haveExtractedBinaries = true
+		return true
+	}
+	// Add binaries to $PATH:
+	if !haveChangedPATH {
+		oldPathEnv := os.Getenv("PATH")
+		newPathEnv := fmt.Sprintf("%s:%s/bin", oldPathEnv, binariesDir)
```

Comment:
It might be better to put binariesDir/bin *before* PATH to make sure that installed binaries are used.

Reply:
Works well locally.